aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -113,27 +140,44 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
   return self.copy_(args.lgamma_().sum(-1).add_(p * (p - 1) * std::log(M_PI) / 4.));
 }
 
+
+Tensor sigmoid(const Tensor& self) {
+  Tensor result = at::empty({0}, self.options());
+  return at::sigmoid_out(result, self);
+}
+Tensor& _sigmoid__cpu(Tensor& self) {
+  return at::sigmoid_out(self, self);
+}
+Tensor& _sigmoid_out_cpu(Tensor& result, const Tensor& self) {
+  auto iter = TensorIterator::unary_op(result, self);
+  sigmoid_stub(iter->device_type(), *iter);
+  return result;
+}
+
 // NB: If you use this macro, you may also need to add a CUDA forwarding
 // stub in CUDAUnaryOps
 
-#define IMPLEMENT_UNARY_OP_VEC(op)                              \
-  Tensor op(const Tensor& self) {                               \
-    Tensor result = at::empty({0}, self.options());             \
-    return at::op##_out(result, self);                          \
-  }                                                             \
-  Tensor& _##op##__cpu(Tensor& self_) {                         \
-    if (self_.numel() > 0) {                                    \
-      Tensor self = sort_strides(self_);                        \
-      op##Impl(kCPU, self, self);                               \
-    }                                                           \
-    return self_;                                               \
-  }                                                             \
-  Tensor& _##op##_out_cpu(Tensor& result, const Tensor& self) { \
-    result.resize_(self.sizes());                               \
-    if (result.numel() > 0) {                                   \
-      op##Impl(kCPU, result, self);                             \
-    }                                                           \
-    return result;                                              \
+#define IMPLEMENT_UNARY_OP_VEC(op)                                         \
+  Tensor op(const Tensor& self) {                                          \
+    Tensor result = at::empty({0}, self.options());                        \
+    return at::op##_out(result, self);                                     \
+  }                                                                        \
+  Tensor& _##op##__cpu(Tensor& self) {                                     \
+    return at::op##_out(self, self);                                       \
+  }                                                                        \
+  Tensor& _##op##_out_cpu(Tensor& result, const Tensor& self) {            \
+    if (self.is_contiguous() && result.is_contiguous()) {                  \
+      result.resize_(self.sizes());                                        \
+      AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), op##_cpu, [&]() {     \
+        vml::v##op(                                                        \
```

Comment:
Currently this means that for users without MKL we won't compile the SLEEF version with avx2. This will immediately be resolved by a follow up patch that creates a separate vml.cpp in cpu to have a standalone set of functions that do their own dispatch and codegen.